core/src/test/scala/com/salesforce/op/stages/impl/feature/DateMapToUnitCircleVectorizerTest.scala

```diff
@@ -0,0 +1,103 @@
+ */
+
+package com.salesforce.op.stages.impl.feature
+
+import com.salesforce.op.features.types._
+import com.salesforce.op.stages.base.sequence.SequenceModel
+import com.salesforce.op.test.{OpEstimatorSpec, TestFeatureBuilder}
+import com.salesforce.op.utils.spark.OpVectorMetadata
+import org.apache.spark.ml.{Estimator, Transformer}
+import org.apache.spark.ml.linalg.Vectors
+import com.salesforce.op.utils.spark.RichDataset._
+import com.salesforce.op.utils.spark.RichMetadata._
+import org.joda.time.{DateTime => JDateTime}
+import org.junit.runner.RunWith
+import org.scalatest.junit.JUnitRunner
+
+@RunWith(classOf[JUnitRunner])
+class DateMapToUnitCircleVectorizerTest extends OpEstimatorSpec[OPVector, SequenceModel[DateMap, OPVector],
+  DateMapToUnitCircleVectorizer[DateMap]] {
+
+  val eps = 1E-4
+  val sampleDateTimes = Seq[JDateTime](
+    new JDateTime(2018, 2, 11, 0, 0, 0, 0),
+    new JDateTime(2018, 11, 28, 6, 0, 0, 0),
+    new JDateTime(2018, 2, 17, 12, 0, 0, 0),
+    new JDateTime(2017, 4, 17, 18, 0, 0, 0),
+    new JDateTime(1918, 2, 13, 3, 0, 0, 0)
+  )
+
+  val (inputData, f1) = TestFeatureBuilder(
+    sampleDateTimes.map(x => Map("a" -> x.getMillis, "b" -> x.getMillis).toDateMap)
+  )
+
+  /**
+   * Estimator instance to be tested
+   */
+  override val estimator = new DateMapToUnitCircleVectorizer[DateMap]().setInput(f1)
+    .setTimePeriod(TimePeriod.HourOfDay)
+  /**
+   * Expected result of the transformer applied on the Input Dataset
+   */
+  override val expectedResult: Seq[OPVector] = sampleDateTimes
+    .map{ v =>
+      val rad = DateToUnitCircle.convertToRandians(Option(v.getMillis), TimePeriod.HourOfDay)
+      (rad ++ rad).toOPVector
+    }
+
+  it should "work with its shortcut as a DateMap" in {
+    val output = f1.toUnitCircle(TimePeriod.HourOfDay)
+    val transformed = output.originStage.asInstanceOf[DateMapToUnitCircleVectorizer[DateMap]]
+      .fit(inputData).transform(inputData)
+    val actual = transformed.collect(output)
+    all (actual.zip(expectedResult).map(g => Vectors.sqdist(g._1.value, g._2.value))) should be < eps
+  }
+
+  it should "work with its shortcut as a DateTimeMap" in {
+    val (inputDataDT, f1DT) = TestFeatureBuilder(
+      sampleDateTimes.map(x => Map("a" -> x.getMillis, "b" -> x.getMillis).toDateTimeMap)
+    )
+    val output = f1DT.toUnitCircle(TimePeriod.HourOfDay)
+    val transformed = output.originStage.asInstanceOf[DateMapToUnitCircleVectorizer[DateMap]]
+      .fit(inputData).transform(inputData)
+    val actual = transformed.collect(output)
+    all (actual.zip(expectedResult).map(g => Vectors.sqdist(g._1.value, g._2.value))) should be < eps
+  }
+
+  it should "make the correct metadata" in {
+    val fitted = estimator.fit(inputData)
+    val meta = OpVectorMetadata(fitted.getOutputFeatureName, fitted.getMetadata())
+    meta.columns.length shouldBe 4
+    meta.columns.map(_.grouping.get) shouldEqual Seq("a", "a", "b", "b")
```

Comment:
`meta.columns.flatMap(_.grouping)` and `meta.columns.flatMap(_.descriptorValue)`